File: Logitude/Logitude.BL/CommonDataModel/DataContracts/ShipmentShareDocumentsData.cs
using Logitude.BL.CommonDataModel.EntityPMs;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Logitude.BL.CommonDataModel.DataContracts
{
    public class ShipmentShareDocumentsData
    {
        public List<ShareDocument> ShareDocuments { get; set; }
        public string ShipmentNumber { get; set; }
        public string EntityId { get; set; }
        public string AgentSharedManifestRef { get; set; }
        public string ShipmentLevelCode { get; set; }
        public string ShipmentLevelName { get; set; }
        public string AgentId { get; set; }
        public int TenantAgent { get; set; }

    }

    public class ShareDocument
    {
        public string DocumentTypeName { get; set; }
        public string DocumentTypeCode { get; set; }
        public bool IsReady { get; set; }
        public DateTime? LastUpdateDate { get; set; }
        public DateTime? LastShareDate { get; set; }
        public string SecurityId { get; set; }
        public string DirectionCode { get; set; }
        public string FileName { get; set; }
        public string DocumentId { get; set; }
        public string DocumentsFilingId { get; set; }
        public string Extension { get; set; }
        public double? FileSize { get; set; }
        public string EntityId { get; set; }

        public string DocumentTypeId { get; set; }
        public string DocumentTypeCopyId { get; set; }
        public DocumentsFilingPM DocumentsFilingPM { get; set; }
        public string DocumentOutId { get; set; }

        public string ActionButtonLabel { get; set; }

    }
}

File: Logitude/AngularModules/AngularModules/Common/DataContracts/ShipmentShareDocumentsData.ts
/// <reference path="../entitypms/documentoutpm.ts" />

import {DocumentOutPM} from '../EntityPMs/DocumentOutPM';
import {DocumentsFilingPM} from '../EntityPMs/DocumentsFilingPM';
export class ShipmentShareDocumentsData {
 
    public ShareDocuments: ShareDocument[];
    public IncludedShareDocuments: ShareDocument[];
    
    public ShipmentNumber: string;
    public  EntityId : string;
    public  AgentSharedManifestRef : string;
    public  ShipmentLevelCode : string;
    public  ShipmentLevelName: string;
    public  AgentId: string;
    public  TenantAgent : number;


}

export class ShareDocument {
    public DocumentTypeName: string;
    public DocumentTypeCode: string;
    public IsReady: boolean;
    public LastUpdateDate: Date;
    public LastShareDate: Date;
    public SecurityId: string;
    public EntityId: string;
    public FileName: string;
    public DirectionCode: string;
    public Included: boolean;
    public ShipmentNumber: string;
    public DocumentId: string;
    public DocumentsFilingId: string;
    public Extension: string;
    public FileSize: number;
    public VisibleBliudDocumentButton: boolean = false;
    public VisibleViewButton: boolean = false;
    public VisibleUploadButton: boolean = false;
    public TemplateFormatCode: string;

    public DocumentTypeId: string;
    public DocumentsFilingPM: DocumentsFilingPM;
    public DocumentOutPM: DocumentOutPM;
    public DocumentTypeCopyId: string;
    public ActionButtonLabel: string;
    public DocumentOutId: string;

}

